nevergrad/parametrization/choice.py

```diff
@@ -0,0 +1,245 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import typing as t
+import numpy as np
+from nevergrad.common.typetools import ArrayLike
+from ..instrumentation import discretization  # TODO move along
+from . import core
+from .container import Tuple
+from .data import Array
+from .data import Scalar
+# weird pylint issue on "Descriptors"
+# pylint: disable=no-value-for-parameter
+
+
+C = t.TypeVar("C", bound="Choice")
+T = t.TypeVar("T", bound="TransitionChoice")
+
+
+class BaseChoice(core.Dict):
+
+    def __init__(self, *, choices: t.Iterable[t.Any], **kwargs: t.Any) -> None:
+        assert not isinstance(choices, Tuple)
+        lchoices = list(choices)  # for iterables
+        super().__init__(choices=Tuple(*lchoices), **kwargs)
+
+    def __len__(self) -> int:
+        """Number of choices
+        """
+        return len(self.choices)
+
+    @property
+    def descriptors(self) -> core.Descriptors:
+        return core.Descriptors(deterministic=self.choices.descriptors.deterministic,
+                                continuous=self.choices.descriptors.continuous)
+
+    @property
+    def index(self) -> int:
+        raise Exception
+
+    @property
+    def choices(self) -> Tuple:
+        """The different options, as a Tuple Parameter
+        """
+        return self["choices"]  # type: ignore
+
+    @property
+    def value(self) -> t.Any:
+        return core.as_parameter(self.choices[self.index]).value
+
+    @value.setter
+    def value(self, value: t.Any) -> None:
+        self._find_and_set_value(value)
+
+    def _find_and_set_value(self, value: t.Any) -> int:
+        index = -1
+        # try to find where to put this
+        nums = sorted(int(k) for k in self.choices._parameters)
+        for k in nums:
+            choice = core.as_parameter(self.choices[k])
+            try:
+                choice.value = value
+            except Exception:  # pylint: disable=broad-except
+                pass
+            else:
+                index = int(k)
+                break
```

Comment:
nit: you can put that inside the `try`? Also the `int` cast is not needed

Reply:
there must be an absolutely minimum number of stuff in the try, so that we don't hide any weird errors (like not being able to cast k to int)
Then again as you say, k is supposedly already an int so that should not be a problem indeed^^
So, fair enough, this is simplified